### PR TITLE
Split AuthenticationController.Authenticate into a POST and a GET

### PR DESCRIPTION
**Pull request by scottbommarito**
The route is used as both GET and POST, so in order for it to function correctly it needs to be split into two methods.

### CODE REVIEW

**Comment by shishirx34** [comment]
Looks good.

File: src/NuGetGallery/Controllers/AuthenticationController.cs [hunk 1]
```diff
@@ -234,9 +234,22 @@ public virtual ActionResult LogOff(string returnUrl)
             return SafeRedirect(returnUrl);
         }
 
+        [ActionName("Authenticate")]
```
Comment (skofman1) [inline src/NuGetGallery/Controllers/AuthenticationController.cs]: >ActionName [](start = 9, length = 10)

There is a build brake in AuthenticationControllerFacts.cs

Comment (scottbommarito) [inline src/NuGetGallery/Controllers/AuthenticationController.cs]: Fixed

**Comment by skofman1** [comment]
:ship:

**Pull request merged**